### ISSUE TITLE
dev-libs/libsodium-1.0.18 fails to build with lto on arm64

### ISSUE THREAD

**Issue by Jannik2099** [InBetweenNames/gentooLTO#456]
Libsodium fails to build with lto, see [libsodium-build.log](https://github.com/InBetweenNames/gentooLTO/files/3992629/libsodium-build.log)

I'm surprised that other users haven't encountered this as libsodium is a dependency of keepassxc, which is mentioned in the ltoworkarounds.conf. Perhaps this is due to the asm useflag since this is where the compile fails, I can try emerging it without the asm useflag tomorrow. I haven't tried any other workarounds besides disabling lto, if you have any ideas I'm happy to test them.

The package compiled succesfully without lto, and for those curious this was the only build fail I encountered on my kde arm64 system so far.

**Comment by elsandosgrande** [comment]
What is keepassxc?

**Comment by Jannik2099** [comment]
https://packages.gentoo.org/packages/app-admin/keepassxc

It's the community edition of keepass, a popular password manager

**Comment by elsandosgrande** [comment]
I have never used it, nor do I have it installed. 

Have you tried building it for x86 (32-bit)? If it succeeds there by some miracle of God, we can narrow the workaround down.

**Comment by Jannik2099** [comment]
I will do that the next time I get access to an x86 machine, somewhere around end of next week

**Comment by Jannik2099** [comment]
Libsodium builds fine with lto on x86, even with the asm useflag.
With the asm useflag on arm64 lto fails, works fine without asm.

Can the overlay filter flags depending on arch?

**Comment by elsandosgrande** [comment]
So... I realize that I misread "arm64" as "amd64" earlier, but this, it seems, was not in vain. Given that it works on 32-bit, I'll try 64-bit on my laptop, though I can't actually test it to see if it works properly, as I will be emerging libsodium on its own.

Could you try ARM 32-bit (not sure what it's called, like how ARM 64-bit is aarch64 from what I recall, assuming that I am not getting my wires crossed) @Jannik2099? Also, are you running on a Raspberry Pi, or... and how did you manage to set it up, since I did not see any official ARM handbook the last time I looked at the handbooks page (the latter one is mostly for my own curiosity, so we should probably move that to the Gitter chat, no?)?

Regarding the filtering of flags per architecture, it is most likely possible, but will take some Bash and `app-portage/portage-bash-mv` know-how. I think that it will look like the LibreOffice line in the workarounds, since it can selectively pick when to override the flags based on the USE flags presented. Given that `ARCH="aarch64"` is probably another extended USE flag, it could be handled in the same manner.

**Comment by Jannik2099** [comment]
Yeah, arm64 is usually called aarch64, but the Gentoo keyword is arm64 so I went with that. 32-bit arm is just called... arm.

With the asm use flag, lto fails on arm aswell. So it seems asm + lto is broken on both arm and arm64 here.

**Comment by InBetweenNames** [comment]
Confirmed!

**Comment by InBetweenNames** [comment]
It would be good to add this as a conditional workaround for `(arm || arm64) && asm`
